ftw/upgrade/command/jsonapi.py

```diff
@@ -33,6 +36,27 @@ def _make_request(self, method, action, site=None, **kwargs):
         return response
 
 
+class TempfileAuth(AuthBase):
+    """A requests authenticator which writes a tempfile with a random
+    hash to verify that the client and the server is at the same
+    machine and with the same user.
+    """
+
+    def __call__(self, request):
+        self._generate_tempfile()
+        value = ':'.join((os.path.basename(self.authfile.name),
+                          self.authhash)).encode('base64')
+        request.headers['x-ftw.upgrade-tempfile-auth'] = value
+        return request
+
+    def _generate_tempfile(self):
+        self.authhash = os.urandom(32).encode('hex')
```

Comment:
Ideally we would not directly use the output from `/dev/[u]random` here as the key, but instead use a [key derivation function](https://docs.python.org/2/library/hashlib.html#key-derivation-function) generate a derived key that has desirable cryptographic properties.